packages/airnode-node/src/api/index.ts

```diff
@@ -287,11 +288,33 @@ export async function callApi(payload: ApiCallPayload): Promise<LogsData<ApiCall
   if (verificationResult) return verificationResult;
 
   const processedPayload = await preProcessApiSpecifications(payload);
-
-  const [logs, response] = await performApiCall(processedPayload);
-  if (isPerformApiCallFailure(response)) {
-    return [logs, response];
+  const ois = payload.config.ois.find((o: OIS) => o.title === payload.aggregatedApiCall.oisTitle) as OIS;
+  const endpoint = ois.endpoints.find((e: Endpoint) => e.name === payload.aggregatedApiCall.endpointName) as Endpoint;
+
+  if (!endpoint.operation && endpoint.fixedOperationParameters.length === 0) {
```

Comment:
This is pretty difficult to follow. Maybe it could be refactored/shortened a bit? Also adjusted the message grammer a little bit. Might also be worth adding some information about the specific invalid endpoint for easier debugging (e.g. the name)

```ts
import isEmpty from 'lodash/isEmpty';

if (!endpoint.operation && isEmpty(endpoint.fixedOperationParameters)) {
  if (isEmpty(endpoint.preProcessingSpecifications) && isEmpty(endpoint.postProcessingSpecifications) {
    const message = `Failed to skip API call. Ensure at least one of 'preProcessingSpecifications' or 'postProcessingSpecifications' is defined and is not an empty array.`;
    const log = logger.pend('ERROR', message);
    return [log, { success: false, errorMessage: message }];
  }
  return processSuccessfulApiCall(payload, { data: processedPayload.aggregatedApiCall.parameters });
}

const [logs, response] = await performApiCall(processedPayload);
if (isPerformApiCallFailure(response)) {
  return [logs, response];
}
return processSuccessfulApiCall(payload, response);
```

Reply:
It's probably also worth adding a comment explaining what's going on here too, since it's not obvious